applications/FluidDynamicsApplication/custom_strategies/strategies/gear_scheme.h

```diff
@@ -741,6 +768,76 @@ class GearScheme : public Scheme<TSparseSpace, TDenseSpace>
             std::cout << "Performed OSS Projection" << std::endl;
     }
 
+    /** On periodic boundaries, the nodal area and the values to project need to take into account contributions from elements on
+     * both sides of the boundary. This is done using the conditions and the non-historical nodal data containers as follows:\n
+     * 1- The partition that owns the PeriodicCondition adds the values on both nodes to their non-historical containers.\n
+     * 2- The non-historical containers are added across processes, transmiting the right value from the condition owner to all partitions.\n
+     * 3- The value on all periodic nodes is replaced by the one received in step 2.
+     */
+    void PeriodicConditionProjectionCorrection(ModelPart& rModelPart)
+    {
+        const int num_nodes = rModelPart.NumberOfNodes();
+        const int num_conditions = rModelPart.NumberOfConditions();
+
+        #pragma omp parallel for
+        for (int i = 0; i < num_nodes; i++) {
+            auto it_node = rModelPart.NodesBegin() + i;
+
+            it_node->SetValue(NODAL_AREA,0.0);
+            it_node->SetValue(ADVPROJ,ZeroVector(3));
+            it_node->SetValue(DIVPROJ,0.0);
+        }
+
+        #pragma omp parallel for
+        for (int i = 0; i < num_conditions; i++) {
+            auto it_cond = rModelPart.ConditionsBegin() + i;
+
+            if(it_cond->Is(PERIODIC)) {
+                ModelPart::ConditionType::GeometryType& rGeom = it_cond->GetGeometry();
+                unsigned int nodes_in_cond = rGeom.PointsNumber();
+
+                double NodalArea = 0.0;
+                array_1d<double,3> AdvProj(3,0.0);
+                double DivProj = 0.0;
+                for ( unsigned int i = 0; i < nodes_in_cond; i++ )
+                {
+                    NodalArea += rGeom[i].FastGetSolutionStepValue(NODAL_AREA);
+                    noalias(AdvProj) += rGeom[i].FastGetSolutionStepValue(ADVPROJ);
+                    DivProj += rGeom[i].FastGetSolutionStepValue(DIVPROJ);
+                }
+
+                for ( unsigned int i = 0; i < nodes_in_cond; i++ )
+                {
+                    /* Note that it is assumed that each node belongs to a single periodic link,
+                        * if this lock is necessary, the algorithm will not work anyways.
```

Comment:
I rephrased it in the updated version, but the idea is that this lock should not be necessary in theory (given the current periodic condition algorithm, which does not support a node being involved in multiple periodic links at the same time), but I prefer to keep it as future-proofing or, in case something goes wrong, to (hopefully) get a visible error somewhere else instead of an OpenMP race condition.